src/Login/ProtectedContent.jsx

```diff
@@ -279,9 +279,16 @@ class ProtectedContent extends React.Component {
     window.scrollTo(0, 0);
     const pageFullWidthClassModifier = isPageFullScreen(this.props.location.pathname) ? 'protected-content--full-screen' : '';
     if (this.state.redirectTo) {
+      let fromURL = '/';
+      if (this.state.from && this.state.from.pathname) {
+        fromURL = this.state.from.pathname;
+        if (this.state.from.search && this.state.from.search !== '') {
```

Comment:
Should we not account for a non-search URL with query parameters like this?
`https://qa-covid19.planx-pla.net/open-JHU/explorer?filter={...}`

Reply:
oh sure! I think this implementation would takes care of that as well. It is called `location.search` but actually it is the the URL query string, not just `?search=`

https://github.com/ReactTraining/history/blob/master/docs/api-reference.md#locationsearch

Reply:
oh!! haha funny coincidence